torch/_inductor/ir.py

```diff
@@ -1215,7 +1215,14 @@ def create_multilayer(
         """
         # TODO(jansel): realize the reduction so we can do dynamic indexing
         reduction_numel = sympy_product(reduction_ranges)
-        block_size = FloorDiv(reduction_numel + (split - 1), split)
+        rnumel_hint, split_hint = V.graph.sizevars.size_hints([reduction_numel, split])
```

Comment:
I wonder is it possible to just replace all sympy symbols in reduction_ranges with 1, calculate sympy_product(reduction_ranges) and check divisibility? In this case, I guess "guard_multiple_of" won't trigger recompilations.

Reply:
I think that check will consider divisibility to be false in most cases? Like reduction_ranges for some tensor of size [s0, s1] is just going to be s0*s1, and replacing `s*` with 1 just gives us a sympy_product of 1

Reply:
Is it possible to distinguish between dynamic shapes and static shapes by using something like `V.graph.sizevars.symbolic_hint()`? For our use cases, the product of static dims are divisible by split. For simplicity I feel we only need to handle this case?